### PR TITLE
First draft of doc explaining how rolling updates work and the zero d…

### PR DESCRIPTION
**Pull request by ollieanwyll**
Connect to issue ministryofjustice/cloud-platform#632

This PR contains a document that explains how Zero Downtime Deployments work in the Cloud Platform and the advantages they bring.

### CODE REVIEW

**Review by vijay-veeranki — APPROVED** [review]

File: 03-other-topics/008-zero-downtime-deployment.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+The new deployment is applied incrementally, one Pod at a time until all Pods are running the latest version.
+
+If an application is exposed publicly, traffic will only be routed to the available Pods.
+
+### The Major Advantage
+
+Where Rolling Updates really brings benefit is the enabling of Continuous Integration and Continuous Delivery.
+
+The ability to constantly update your application with micro-changes, with zero downtime, brings a host of benefits.
```
Comment (jasonBirchall) [inline 03-other-topics/008-zero-downtime-deployment.md]: Doesn't read well, are you able to change?
```with micro-changes, with zero downtime```

File: 03-other-topics/008-zero-downtime-deployment.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+---
+category: cloud-platform
+expires: 2019-08-12
+---
+
+# Zero Downtime Deployments
+
+## Introduction
+
+In the past at the MoJ, application team members were used to application service outage when deploying a new version of their application.
```
Comment (jasonBirchall) [inline 03-other-topics/008-zero-downtime-deployment.md]: Can you please rewrite this paragraph?

File: 03-other-topics/008-zero-downtime-deployment.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+A rolling update works by ensuring there is always one extra Pod than the maximum number stated in the deployment.
+
+The new deployment is applied incrementally, one Pod at a time until all Pods are running the latest version.
+
+If an application is exposed publicly, traffic will only be routed to the available Pods.
+
+### The Major Advantage
+
+Where Rolling Updates really brings benefit is the enabling of Continuous Integration and Continuous Delivery.
```
Comment (jasonBirchall) [inline 03-other-topics/008-zero-downtime-deployment.md]: Remove the capitals in Rolling and Update.

File: 03-other-topics/008-zero-downtime-deployment.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+## Rolling Updates
+
+Rolling updates introduce the ability to update an application without any downtime.
+
+A rolling update works by ensuring there is always one extra Pod than the maximum number stated in the deployment.
+
+The new deployment is applied incrementally, one Pod at a time until all Pods are running the latest version.
+
+If an application is exposed publicly, traffic will only be routed to the available Pods.
```
Comment (alkar) [inline 03-other-topics/008-zero-downtime-deployment.md]: This is only true if and only if the user configured `readinessProbes` correctly. This is a large topic so perhaps provide a link to the readiness/health probes?

File: 03-other-topics/008-zero-downtime-deployment.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+With the Cloud Platform, this is now an annoyance of the past, thanks to Rolling Updates.
+
+## Rolling Updates
+
+Rolling updates introduce the ability to update an application without any downtime.
+
+A rolling update works by ensuring there is always one extra Pod than the maximum number stated in the deployment.
+
+The new deployment is applied incrementally, one Pod at a time until all Pods are running the latest version.
```
Comment (alkar) [inline 03-other-topics/008-zero-downtime-deployment.md]: This is a good example but in reality the actual numbers are configurable (ie. you can get it to say replace two pods at a time). Can we provide a link to the docs for how to configure the rolling update strategy?

**Review by alkar — CHANGES_REQUESTED** [review]

**Review by jasonBirchall — APPROVED** [review]

**Review by alkar — APPROVED** [review]

**Pull request merged**